src/control/server/server.go

```diff
@@ -356,7 +356,25 @@ func (srv *server) start(ctx context.Context, shutdown context.CancelFunc) error
 		shutdown()
 	}()
 
-	return errors.Wrapf(srv.harness.Start(ctx, srv.sysdb, srv.pubSub, srv.cfg),
+	drpcSetupReq := &drpcServerSetupReq{
```

Comment:
I remember I was asked to keep this setup code out of the server.go file, just fyi. Also it may introduce difficulties with restarting ranks (system start and stop) if the cleanup and restart is not performed within the engine start/run loops.

Reply:
This is the daos_server socket, not the per-engine socket. There's only one setup/teardown per lifetime of the server, right? Not sure how that would cause problems for rank restart.

I moved this out here because it kept things a bit cleaner (didn't require interfaces to be changed to handle the database), but I'll take a look at whether or not this is really necessary.